src/lib.rs

```diff
@@ -250,7 +257,7 @@ impl Logger {
 
 impl Log for Logger {
     fn enabled(&self, metadata: &Metadata) -> bool {
-        self.filter.enabled(metadata)
+        self.filter.load().enabled(metadata)
```

Comment:
Similarly, the object lives only as long as `enabled()` runs, avoiding unnecessary guards.